SystemManager/Inc/SM.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * SM.hpp
+ *
+ * System Manager Header
+ *
+ * Created on: Nov 7, 2022
+ * Author(s): Gordon Fountain
+ */
+
+#ifndef ZPSW3_SM_HPP
+#define ZPSW3_SM_HPP
+
+#include <stdint.h>
+#include "cmsis_os.h"
+// #include "LOS_Link.hpp"
+// #include "LOS_Actuators.hpp"
+#include "SM_StateManager.hpp"
+
+class SystemState;
+
+namespace SM {
+    // Enumerates the current drone status
+    enum _System_Manager_Drone_Operation_Mode{BOOT, DISARMED, GROUND_OPS, FLIGHT, FATAL_FAILURE = -1};
```

Comment:
nit: do we have to start with an underscore? I thought it was generally discouraged.

Reply:
I'm unfamiliar with how enums work, so I was curious about the = -1 at the end.

Reply:
Enums (enumerations) just assign a number to each of those names in the SM namespace. It defaults to starting at 0 (so BOOT == 0), but you can overwrite the values. I overwrote the FATAL_FAILURE one to be a negative number so it can be easily identified if we need to.